core/services/pipeline/orm.go

```diff
@@ -427,14 +378,12 @@ func (o *orm) ResultsForRun(ctx context.Context, runID int64) ([]Result, error)
 	var results []Result
 	err = postgres.GormTransaction(ctx, o.db, func(tx *gorm.DB) error {
 		var resultTaskRun TaskRun
-		err = tx.
-			Preload("PipelineTaskSpec").
-			Joins("INNER JOIN pipeline_task_specs ON pipeline_task_runs.pipeline_task_spec_id = pipeline_task_specs.id").
-			Where("pipeline_run_id = ?", runID).
-			Where("finished_at IS NOT NULL").
-			Where("pipeline_task_specs.successor_id IS NULL").
-			Where("pipeline_task_specs.dot_id = ?", ResultTaskDotID).
-			First(&resultTaskRun).
+		err = tx.Raw(`
+			SELECT * FROM pipeline_task_runs
+			WHERE pipeline_run_id = ? 
+				AND finished_at IS NOT NULL
+				AND dot_id = ?
+`, runID, ResultTaskDotID).Scan(&resultTaskRun).
```

Comment:
Honestly this query should just fetch the result on the `pipeline_run`.

That `__result__` task needs to go.

Reply:
agreed, I think we should do it outside this change though